plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1728,7 +1729,9 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
             IcebergColumnHandle columnHandle = (IcebergColumnHandle) column;
             // Iceberg metadata columns can not be used to filter a table scan in Iceberg library
             // TODO (https://github.com/trinodb/trino/issues/8759) structural types cannot be used to filter a table scan in Iceberg library.
-            if (isMetadataColumnId(columnHandle.getId()) || isStructuralType(columnHandle.getType())) {
+            if (isMetadataColumnId(columnHandle.getId()) || isStructuralType(columnHandle.getType()) ||
+                    // Iceberg orders UUID values differently than Trino (perhaps due to https://bugs.openjdk.org/browse/JDK-7025832), so allow only IS NULL / IS NOT NULL checks
+                    (columnHandle.getType() == UUID && !(domain.isOnlyNull() || domain.getValues().isAll()))) {
```

Comment:
Does an equality predicate work? Feels like asking for a range of UUIDs would be unusual but asking for rows matching a specific UUID would be common

Reply:
Equality is consistent between Trino and Java's UUID, but even for equality predicate the ordering matters when you match the searched-for value with file min/max information.